File: server/worker/src/database/projections/valut.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;
use value::Value;

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Valut {
    pub id: Uuid,
    pub created_at: DateTime<Utc>,
    pub last_modification_at: DateTime<Utc>,
    pub user_id: Uuid,
    pub asset_id: Uuid,
    pub balance: Value,
}

File: web/packages/types/valut.ts
import { z } from "zod";
import { Uuid } from "./primitives/uuid";
import { Datetime } from "./primitives/datetime";
import { Value } from "./primitives/value";

export const Valut = z.object({
  id: Uuid,
  created_at: Datetime,
  last_modification_at: Datetime,
  user_id: Uuid,
  asset_id: Uuid,
  balance: Value,
});
export type Valut = z.infer<typeof Valut>;
